SPEC.md

```diff
@@ -848,23 +849,39 @@ Single- and double-quotes do not need to be escaped within a multi-line string.
 
 A `File` or `Directory` declaration may have have a string value indicating a relative or absolute path on the local file system.
 
-Within a WDL file, literal values for files may only be (relative or absolute) paths that are local to the execution environment. If the specified path does not exist, it is an error unless the declaration is optional.
-
 ```wdl
 task literals_paths {
   input {
-    # If the user does not overide the value of `f1`, and /foo/bar.txt
-    # does not exist, it is an error.
     File f1 = "/foo/bar.txt"
-
-    # If the user does not override the value of `f2` and /foo/bar.txt
-    # does not exist, then `f2` is set to `None`.
-    File? f2 = "/foo/bar.txt"
+    File? f2
   }
+
+  command <<<
+    # If the user does not overide the value of `f1`, and /foo/bar.txt
+    # does not exist, an error will occur when the file is accessed here.
+    cat "~{f1}"
+
+    # If the user does not specify the value of `f2` it's value is `None`,
+    # which results in the empty-string when interpolated. `-f ""` is
+    # always false.
+    if [ -f "~{f2}" ]; then
+      echo "~{f2}"
+    fi
+  >>>
 }
 ```
 
-An execution engine may support [other ways](#input-and-output-formats) to specify `File` and `Directory` inputs (e.g., as URIs), but prior to task execution it must [localize inputs](#task-input-localization) so that the runtime value of a `File`/`Directory` variable is a local path.
+Within a WDL file, literal values for files and directories may only be paths that are local to the execution environment.
```

Comment:
So an engine is not allowed to support something like:
```
inputs {
    File genome_reference = "https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz"
}
```
?

Reply:
Good point - I've updated the spec to say that the execution engine is only required to support paths, but I'm open to expanding this in WDL 1.3 to require engines to support http(s) URLs as well. This would be consistent with the requirement to support http(s) URLs in imports.